homeassistant/components/directv/media_player.py

```diff
@@ -116,62 +123,75 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
                 _LOGGER.debug(
                     "Discovered device %s on host %s with "
                     "client address %s is already "
                     "configured",
-                    str.title(loc["locationName"]),
+                    loc_name,
                     host,
                     loc["clientAddr"],
                 )
             else:
                 _LOGGER.debug(
                     "Adding discovered device %s with client address %s",
-                    str.title(loc["locationName"]),
+                    loc_name,
                     loc["clientAddr"],
                 )
-                hosts.append(
-                    [
-                        str.title(loc["locationName"]),
-                        host,
-                        DEFAULT_PORT,
-                        loc["clientAddr"],
-                    ]
+                dtv = DIRECTV(host, DEFAULT_PORT, loc["clientAddr"])
+                devices.append(
+                    DirecTvDevice(
+                        loc_name, loc["clientAddr"], dtv, discovery_dtv_version,
+                    )
                 )
+                known_devices.add((host, loc["clientAddr"]))
 
-    dtvs = []
+    add_entities(devices)
 
-    for host in hosts:
-        dtvs.append(DirecTvDevice(*host))
-        hass.data.setdefault(DATA_DIRECTV, set()).add((host[1], host[3]))
 
-    add_entities(dtvs)
+def _get_receiver_version(client):
+    """Return the version of the DirectTV receiver."""
+    try:
+        return client.get_version()
+    except requests.exceptions.RequestException as ex:
+        _LOGGER.debug("Request exception %s trying to get receiver version", ex)
+        return None
 
 
 class DirecTvDevice(MediaPlayerDevice):
     """Representation of a DirecTV receiver on the network."""
 
-    def __init__(self, name, host, port, device):
+    def __init__(self, name, device, dtv, version_info=None):
         """Initialize the device."""
-
-        self.dtv = DIRECTV(host, port, device)
+        self.dtv = dtv
+        self._client_addr = device
         self._name = name
+        self._unique_id = None
         self._is_standby = True
         self._current = None
         self._last_update = None
         self._paused = None
         self._last_position = None
         self._is_recorded = None
-        self._is_client = device != "0"
+        self._is_client = self._client_addr != "0"
         self._assumed_state = None
         self._available = False
         self._first_error_timestamp = None
+        self._receiver_id = None
+        self._version_info = version_info
+
+        if self._version_info:
+            self._receiver_id = "".join(self._version_info.get("receiverId").split())
+            self._unique_id = f"{self._receiver_id}-{self._client_addr}"
 
         if self._is_client:
-            _LOGGER.debug("Created DirecTV client %s for device %s", self._name, device)
+            _LOGGER.debug(
```

Comment:
Please revert this change.

Reply:
A separate clean up PR is welcome where this logging is moved or removed. We don't want side effects in init method.